### ISSUE TITLE
Change analysis_method.algorithms.name to improve usability

### ISSUE THREAD

**Issue by dcferreira** [CN-TU/nta-meta-analysis-specification#25]
The current inputs for ``analysis_method.algorithms.name`` and ``analysis_method.algorithms.subname`` are not consistent enough to be usable for meta-analysis.

A sample of ``analysis_method.algorithms.name``:

| analysis_method.algorithms.name                                  | nr of occurences |
|------------------------------------------------------------------|------------------|
| Nearest Neighbor                                                 | 4                |
| KNN                                                              | 1                |
| Autoencoder                                                      | 2                |
| neural network                                                   | 1                |
| Support Vector Machine (SVM)                                     | 23               |
| Support Vector Machines                                          | 1                |
| Naı̈ve Bayes with kernel density estimation and FCBF prefiltering | 1                |
| Naı̈ve Bayes with kernel density estimation                       | 1                |
| cross-validation                                                 | 1                |
| Xfold-cross-validation                                           | 1                |

These should be easily grouped together.
Right now, there are 81 unique names, which makes it not very useful for meta-analysis.

Additionally, the ``subname``s sometimes contain the same information as the name, and sometimes extend it.

So we need to address 2 problems:
1. The ``name`` field is not useful as it is
2. It's not clear what the ``subname`` should contain (e.g., if the method is "Naı̈ve Bayes with kernel density estimation", and ``name`` is "Naı̈ve Bayes", should ``subname`` be "Naı̈ve Bayes with kernel density estimation" or "with kernel density estimation"

**Comment by dcferreira** [comment]
Talking with @felixig, we have a proposal solution.

For 1:
The idea would be to transform it in a field with a fixed set of possible values.
We would also rename it to something more fitting, like ``family``.
We are preparing a mapping from the current names to this ``family``.

For 2:
Additionally, we would rename ``subname`` to ``detail`` (to be consistent with ``tools``).
This field would have free text, which would contain the full name.
That is, in the "Naı̈ve Bayes" example above, this ``detail`` would be "Naı̈ve Bayes with kernel density estimation".

**Comment by dcferreira** [comment]
@felixig has identified the following families in the current database:
```
autoregressive_models
bayesian
cluster_validation
crossvalidation
decision_tree
ensemble
entropy_based
fuzzy_clustering
glm_regression
graph_modeling
hierarchical_clustering
kmeans_clustering
kmedoids_clustering
knn
markov_process
mixture_model
neural_networks
parameter_search
pca
random_forest
rule_extraction
signature
statistics
stream_clustering
svm
two_step_clustering
wavelet_transform
```

Additionally, he has decided we should change the definition of ``analysis_method.algorithms.type`` to:
> It refers to the normal use of the defined algorithms, i.e., the algorithm is generally considered as a method for <type_value>. Please consider carefully the following default labels (values)

together with the following types:
- classification
The algorithm is supervised and originally intended to perform classification and class-prediction. Based on a labeled set of training data, the algorithm predicts the labels of a set of test data.

- regression
Similar to classification, the algorithm is supervised but, instead of predicting or guessing a class/label, it outputs a predicted numerical value.

- clustering
The algorithm is unsupervised and originally intended to find clusters within the data. Results are based on the intrinsic properties of the analyzed data and the input space drawn by such data (without resorting to any external partition or knowledge)

- modeling
Based on a set of given assumptions and pre-knowledge, the algorithm uses some training data to create a model that summarizes a given process. The difference with other algorithm types--which might and generally do also use models--is that here the "model-according-to-some-initial-assumptions" is the focus of the process (i.e., knowledge discovery), whereas other algorithms are focused on the predicted values (i.e., problem solving), being the model a mean or secondary otuput.

- outlier_detection
The algorithm is unsupervised and originally created to rank data points based on their outlierness properties when compared with the rest of data points in the same dataset. Clustering algorithms can be used for outlier detection, but purely outlier detection algorithms are not forced to create a cluster-like representation of the data.

- space_transformation
The algorithm transforms the input space into an output space that simplifies subsequent analysis, improves the visualization or understanding of the data, or removes constraints.

- specific_detection
The algorithm is specifically crafted to capture, detect, or identify a specific type of phenomenon. These algorithms are usually based on heuristics and ad-hoc steps.

- validation_optimization
The algorithm contributes to validate or optimize the analysis process. Such algorithms validate the outcomes of other algorithms, performs the search of optimal parameters, or nest algorithms to enhance some properties.

**Comment by dcferreira** [comment]
[This file](https://github.com/CN-TU/nta-meta-analysis-specification/files/2566567/names_subnames_replacements.txt) contains a mapping for the current papers in the database, from the old names/subnames to the new family/details/type.
